File: UtilityService/Domain/Entities/Order.cs
namespace UtilityService.Domain.Entities;

public class Order
{
    public Guid Id { get; set; }
    public Guid AuctionId { get; set; }
    public string BuyerUsername { get; set; } = string.Empty;
    public string SellerUsername { get; set; } = string.Empty;
    public string ItemTitle { get; set; } = string.Empty;
    public decimal WinningBid { get; set; }
    public decimal TotalAmount { get; set; }
    public decimal? ShippingCost { get; set; }
    public decimal? PlatformFee { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.PendingPayment;
    public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Pending;
    public string? PaymentTransactionId { get; set; }
    public string? ShippingAddress { get; set; }
    public string? TrackingNumber { get; set; }
    public string? ShippingCarrier { get; set; }
    public DateTimeOffset? PaidAt { get; set; }
    public DateTimeOffset? ShippedAt { get; set; }
    public DateTimeOffset? DeliveredAt { get; set; }
    public string? BuyerNotes { get; set; }
    public string? SellerNotes { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
}

public enum OrderStatus
{
    PendingPayment,
    PaymentReceived,
    Processing,
    Shipped,
    Delivered,
    Completed,
    Cancelled,
    Disputed,
    Refunded
}

public enum PaymentStatus
{
    Pending,
    Processing,
    Completed,
    Failed,
    Refunded
}

File: Web/types/order.ts
export enum OrderStatus {
    PendingPayment = 'PendingPayment',
    PaymentReceived = 'PaymentReceived',
    Processing = 'Processing',
    Shipped = 'Shipped',
    Delivered = 'Delivered',
    Completed = 'Completed',
    Cancelled = 'Cancelled',
    Disputed = 'Disputed',
    Refunded = 'Refunded',
}

export enum PaymentStatus {
    Pending = 'Pending',
    Processing = 'Processing',
    Completed = 'Completed',
    Failed = 'Failed',
    Refunded = 'Refunded',
}

export interface Order {
    id: string;
    auctionId: string;
    buyerId: string;
    buyerUsername: string;
    sellerId: string;
    sellerUsername: string;
    itemTitle: string;
    itemImageUrl?: string;
    winningBid: number;
    totalAmount: number;
    shippingCost?: number;
    platformFee?: number;
    status: OrderStatus;
    paymentStatus: PaymentStatus;
    paymentTransactionId?: string;
    shippingAddress?: string;
    trackingNumber?: string;
    shippingCarrier?: string;
    paidAt?: string;
    shippedAt?: string;
    deliveredAt?: string;
    buyerNotes?: string;
    sellerNotes?: string;
    createdAt: string;
    updatedAt?: string;
    hasReview?: boolean;
}

export interface OrderSummary {
    totalOrders: number;
    pendingPayment: number;
    awaitingShipment: number;
    shipped: number;
    completed: number;
    totalRevenue: number;
}

export interface UpdateOrderStatusDto {
    status: string;
    trackingNumber?: string;
    shippingCarrier?: string;
    notes?: string;
}

export interface ShipOrderDto {
    trackingNumber: string;
    carrier: string;
}
